lib/mongodb.js

```diff
@@ -1606,8 +1606,6 @@ MongoDB.prototype.update = MongoDB.prototype.updateAll = function updateAll(
   const idName = this.idName(modelName);
 
   where = self.buildWhere(modelName, where, options);
-
-  delete data[idName];
```

Comment:
I think we should clone `data`, excluding `id`.

Reply:
Why clone? `data` could be deeply nested, and similar operations are performed in other places. It will introduce a new level of complexity and add to the scope.

`data` is passed around multiple layers and maybe be referenced from a controller, for example in the reported issue.

Reply:
For update to an existing document in mongodb, we delete the id on purpose IIRC. The current code changes the original input as complained by https://github.com/strongloop/loopback-next/issues/3267. I think the fix should leave `data` intact, but remove `id` from the object to be written to the database.

Reply:
@bajtos @b-admike @jannyHou thoughts?

Deleting the `id` is ideal and works perfectly, unless `data` is referenced in a controller. In the controller `data` exists as an instance of the Model, but it is the same object.

So, I guess we should go ahead with deep copy of `data` without the `id`.

Reply:
If we only have to remove `id`, shadow copy is probably good enough.